services/blockchain-indexer/shared/dataService/indexStatus.js

```diff
@@ -60,9 +47,29 @@ const getIndexStatus = async () => {
 	};
 };
 
+const indexStatUpdateListener = async _indexStats => {
+	indexStatsCache = _indexStats;
+	lastUpdate = getCurrentTimestamp();
+
+	const indexStatus = await getIndexStatus();
+	Signals.get('updateIndexStatus').dispatch(indexStatus);
+};
+
+const indexingProgressListener = numJobsInProgress => {
+	isIndexingInProgress = numJobsInProgress > 0;
+	lastUpdate = getCurrentTimestamp();
+};
+
+Signals.get('indexStatUpdate').add(indexStatUpdateListener);
+Signals.get('numJobsInProgressUpdate').add(indexingProgressListener);
+
 const isBlockchainFullyIndexed = () => Number(indexStatsCache.percentage) === 100;
 
 module.exports = {
 	getIndexStatus,
 	isBlockchainFullyIndexed,
+
+	// Testing
```

Comment:
what is this testing?

Reply:
The below functions are exported for unit tests